### ISSUE TITLE
Space home's search can't find DMs

### ISSUE THREAD

**Issue by HarHarLinks** [vector-im/element-meta#2235]
### Steps to reproduce

1. Create (private) space
2. Explicitly add some DMs through the "Add existing room" feature
3. Try to search to the DMs by its (autogenerated, based on the other user) name

#### What did you expect?
find the DM, which looks the same as a room and is shown in the space home
#### What happened instead?
DM is not found

### Operating system

arch

### Browser information

Firefox 119

### URL for webapp

develop.element.io

### Application version

Element version: 2ee082652a18-react-33d527ee5e05-js-8ef2ca681c59 Crypto version: Olm 3.2.15

### Homeserver

matrix.org

### Will you send logs?

No

**Comment by t3chguy** [comment]
That hierarchy view is built by the server `/hierarchy` API

**Comment by HarHarLinks** [comment]
Something makes the DMs appear in the list, so there is a discrepancy between just looking and using the search here.

**Comment by t3chguy** [comment]
Yes, the room list is built by the client until we switch to Sliding Sync where it'll also be built by the server, mixing the two sources wouldn't be sane and consistent.
